### ISSUE TITLE
defaultColDef with filterParams override all colDef.filterParams 

### ISSUE THREAD

**Issue by wanmifam** [ag-grid/ag-grid#4107]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```
**Providing a Reproducible Scenario**
When define defaultColDef with filterParams as below:
```javascript
defaultColDef = {
      editable: true,
      sortable: true,
      filterParams: {
        buttons: ['reset'],
      },
      flex: 1,
      minWidth: 100,
      filter: true,
      resizable: true,
    };
```
It will affect all filterParams of ColDef as below:
```javascript
columnDefs = [
      {
        field: 'row',
        filter: 'agNumberColumnFilter',
        filterParams: {
          filterOptions: ['equals'],
        },
      },
      {
        field: 'name',
        filter: 'partialMatchFilter',
      },
      {
        field: 'name2',
        headerName: 'Name 2',
        filterParams: {
          filterOptions: ['contains'],
        },
      },
    ];
```
- The below ag-Grid version: 23.2.1 isn't affected.
- A plunker
https://plnkr.co/edit/wMO7qbFmnBcf74zR?preview

* **ag-Grid version:** 24.0.0

* **Browser:** Google Chrome, Version 85.0.4183.121

* **Language:** [all]

**Comment by hadassegal** [comment]
This happened to me as well yesterday when I upgraded to version 24.0.0.
Specifically when I tried adding these filterParams to a specific column:
```
 filterParams: {
defaultJoinOperator: 'OR'
}
     
     
```

**Comment by makinggoodsoftware** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by kiril-matev** [comment]
Hello!

Thank you for reporting this issue.

This same issue was reported earlier and tracked by bug:
**AG-4597 Providing default filterParams causes per column filterParams to be shared for all columns**

This bug was fixed as part of ag-Grid v24.1, which was released today.

Please get ag-Grid v24.1 and confirm this is resolved.

**Comment by wanmifam** [comment]
Hello team,

I  confirm this issue is resolved in ag-Grid v24.1.

Thank you.
